server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java

```diff
@@ -547,24 +554,58 @@ public void testCacheFilesAreClosedAfterUse() throws Exception {
         final Client client = client();
 
         internalCluster().ensureAtLeastNumSearchAndDataNodes(numReplicasIndex + 1);
-        createIndexWithDocsAndEnsureGreen(1, 100, indexName);
+        createIndexWithDocsAndEnsureGreen(numReplicasIndex, 100, indexName);
         createRepositoryWithSettings(null, repoName);
         takeSnapshot(client, snapshotName, repoName, indexName);
         restoreSnapshotAndEnsureGreen(client, snapshotName, repoName);
         assertDocCount(restoredIndexName, 100L);
 
+        // Get the available NodeEnvironment instances
+        Iterable<NodeEnvironment> nodeEnvironments = internalCluster().getInstances(NodeEnvironment.class);
+
+        // Filter out search NodeEnvironment(s) since FileCache is initialized only on search nodes and
+        // collect the path for all the cache locations on search nodes.
+        List<Path> searchNodeFileCachePaths = StreamSupport.stream(nodeEnvironments.spliterator(), false)
+            .filter(nodeEnv -> nodeEnv.fileCache() != null)
+            .map(nodeEnv -> nodeEnv.fileCacheNodePath().fileCachePath)
+            .collect(Collectors.toList());
+
+        for (Path fileCachePath : searchNodeFileCachePaths) {
```

Comment:
Can this be refactored into a method like `assertCacheDirectoryCount(int)` or something similar?

Reply:
Sure.